### DISCUSSION TITLE
Create Appearance preference settings

### DISCUSSION THREAD

**Discussion by jandubois** [rancher-sandbox/rancher-desktop#3562] [Research]
We should make app appearance configurable:

### Theme: (*) System   ( ) Light   ( ) Dark

System follows the system settings (what we do right now), Light or Dark would always use the specified theme, regardless of system settings. [^1]

### Accent Color: (*) Rancher Desktop   ( ) System   ( ) Custom

The "Rancher Desktop" accent color should match the logo color. Custom colour will require a colour picker, similar to the "Branding" option in Rancher Manager.

See #3274 for some more discussion of using accent colours.

These settings could go on their own "Application | Appearance" tag, or be combined with other settings on a "General" tab.

[^1]: We need to decide if using the non-system settings only apply to the window contents, or also to the window title bar, menu, and borders.

**Comment by mook-as** [comment]
It's unclear to me why we need a mode that _doesn't_ match the system preferences; could you elaborate?

Note that some parts of the UI (e.g. dashboard) isn't directly controlled by us (we can [override](https://www.electronjs.org/docs/latest/api/native-theme#nativethemethemesource) it if we _really_ want to, probably).

**Reply by jandubois** [reply]
> It's unclear to me why we need a mode that _doesn't_ match the system preferences; could you elaborate?

A user may want to have the Light theme in general, but still prefer the dark theme for RD. Or configure the system to change according to day/night time, but still wants a specific theme for RD all the time.

Same thing for accent colour.

> Note that some parts of the UI (e.g. dashboard) isn't directly controlled by us (we can [override](https://www.electronjs.org/docs/latest/api/native-theme#nativethemethemesource) it if we _really_ want to, probably).

Yes, we can do that. We do this for the screenshots script, but I guess this is now redundant on macOS and Windows, because we also toggle the system theme, so the title and menu bar colors are correct for the theme.
